src/main/java/com/radixdlt/serialization/Serialization.java

```diff
@@ -73,25 +75,43 @@ public static Serialization create(SerializerIds idLookup, SerializationPolicy p
 		availableOutputs.remove(Output.NONE);
 
 		noneProvider = filterProviderFor(ImmutableMap.of());
-		Map<Class<?>, ImmutableSet<String>> allFields =  availableOutputs.stream()
+		Map<Class<?>, ImmutableSet<String>> allFields = availableOutputs.stream()
 				.flatMap(output -> policy.getIncludedFields(output).entrySet().stream())
 				.collect(Collectors.groupingBy(Map.Entry::getKey, flatMapping(e -> e.getValue().stream(), ImmutableSet.toImmutableSet())));
 		allProvider = filterProviderFor(ImmutableMap.copyOf(allFields));
 
 		ImmutableMap.Builder<Output, JacksonCborMapper> dsonBuilder = ImmutableMap.builder();
-		for (Output e : availableOutputs) {
-			dsonBuilder.put(e, JacksonCborMapper.create(idLookup, filterProviderFor(policy.getIncludedFields(e)), true));
-		}
+
+		JacksonCborMapper hashDsonMapper = JacksonCborMapper.create(idLookup, filterProviderFor(policy.getIncludedFields(Output.HASH)), true);
+
+		JacksonCborMapper apiDsonMapper = JacksonCborMapper.create(idLookup, filterProviderFor(policy.getIncludedFields(Output.API)),
+				true, Optional.of(new ApiSerializationModifier(hashDsonMapper)));
+
+		dsonBuilder.put(Output.HASH, hashDsonMapper);
+		dsonBuilder.put(Output.API, apiDsonMapper);
 		dsonBuilder.put(Output.NONE, JacksonCborMapper.create(idLookup, noneProvider, true));
 		dsonBuilder.put(Output.ALL, JacksonCborMapper.create(idLookup, allProvider, true));
+		for (Output e : Lists.newArrayList(Output.PERSIST, Output.WIRE)) {
+			dsonBuilder.put(e, JacksonCborMapper.create(idLookup, filterProviderFor(policy.getIncludedFields(e)), true));
+		}
+
 		dsonMappers = dsonBuilder.build();
 
 		ImmutableMap.Builder<Output, JacksonJsonMapper> jsonBuilder = ImmutableMap.builder();
-		for (Output e : availableOutputs) {
-			jsonBuilder.put(e, JacksonJsonMapper.create(idLookup, filterProviderFor(policy.getIncludedFields(e)), false));
-		}
+
+		JacksonJsonMapper hashJsonMapper = JacksonJsonMapper.create(idLookup, filterProviderFor(policy.getIncludedFields(Output.HASH)), false);
+
+		JacksonJsonMapper apiJsonMapper = JacksonJsonMapper.create(idLookup, filterProviderFor(policy.getIncludedFields(Output.API)),
+				false, Optional.of(new ApiSerializationModifier(hashDsonMapper)));
+
+		jsonBuilder.put(Output.HASH, hashJsonMapper);
+		jsonBuilder.put(Output.API, apiJsonMapper);
 		jsonBuilder.put(Output.NONE, JacksonJsonMapper.create(idLookup, noneProvider, false));
 		jsonBuilder.put(Output.ALL, JacksonJsonMapper.create(idLookup, allProvider, false));
+		for (Output e : Lists.newArrayList(Output.PERSIST, Output.WIRE)) {
```

Comment:
Can just use Lists.asList

Reply:
it takes an array as 2nd arg

Reply:
Actually takes an ellipsis `Arrays.asList(T... items)` which the compiler turns into an array at the call site.

Reply:
:facepalm: you're right. I was looking at `Lists.asList`, not `Arrays`

Reply:
Just reread.  I think @talekhinezh meant `Arrays.asList(...)`, rather than `Lists.asList(...)`, which does something else.